### DISCUSSION TITLE
Runnig iOS UI tests on CI and empty XCResult path on failure

### DISCUSSION THREAD

**Discussion by WojciechTrzasko** [fastlane/fastlane#20529] [General]
Hi Boys and Girls,

I need some help ;)
Generally, I'm trying to set up XCUITest on the CI. My goal is to run the test plan and later publish the XCResult file as an artifact so that other people would be able to download it and check the details of the run.

I'm getting the result file through the `SharedValues::SCAN_GENERATED_XCRESULT_PATH` value and everything works fine when all the tests are green. When any test failures, the `SharedValues::SCAN_GENERATED_XCRESULT_PATH` is empty, which at the moment doesn't fulfill my needs, as the case of failure is the most interesting one and having the ability to check details is a must-have for me.

Do you have any idea why this is happening? Did you have similar problems in the past? 
Here is how I run the `scan`:
```
scan(
  clean: true,      
  workspace: "./ProjectName.xcworkspace",
  scheme: "ProjectName",
  configuration: "Tests",
  testplan: "UITests",
  devices: ["iPhone 13"]
)
```

Any idea would be helpful ;)

**Comment by aokj4ck** [comment]
@WojciechTrzasko I recommend to work around this by 

1. Adding `scan(result_bundle: true)` which changes the SCAN_GENERATED_XCRESULT_PATH from a DerivedData non-deterministic path to a local `fastlane/test_outputs/$SCHEME.xcresult` path
2. Uploading `fastlane/test_outputs/$SCHEME.xcresult`

Check the documentation > parameters > result_bundle https://docs.fastlane.tools/actions/scan/

Also, note that your CI may exit immediately after a failed `scan` suite. You may need to add a configuration to _always_ run your fastlane/test_outputs artifact handling regardless of the earlier steps success/failure.

**Comment by djups** [comment]
@WojciechTrzasko Hmm, not sure, however im using (rescue => ex) after scan. Keep in mind if test fails, and you have scripts/logic after - it would not be called. Pseudo code:

```
scan(

    )
    rescue => ex
      message:"Something went wrong",
      UI.user_error!(ex)
    else
        message:"Result UI Test",
    end
```
